rts/Sim/Units/UnitTypes/Builder.cpp

```diff
@@ -774,20 +787,13 @@ bool CBuilder::StartBuild(BuildInfo& buildInfo, CFeature*& feature, bool& inWait
 			// of the buildee's yardmap, fallback check
 			if (u == nullptr)
 				u = CGameHelper::GetClosestFriendlyUnit(nullptr, buildInfo.pos, buildDistance, allyteam);
+			else {
```

Comment:
why change to `else`? now the line above (`u = GetClosestFriendly`) is a no-op.
```c
if (u == nullptr)
  u = CGameHelper::GetClosestFriendlyUnit(...);
if (u != nullptr) // not equivalent to `else` because the previous line can set a non-null fallback!
```